tests/testSpherePoint.py

```diff
@@ -0,0 +1,751 @@
+            if point1.isFinite():
+                self.assertTrue(point1 == point1)
+                self.assertFalse(point1 != point1)
+
+                pointCopy = copy.deepcopy(point1)
+                self.assertIsNot(pointCopy, point1)
+                self.assertEqual(pointCopy, point1)
+                self.assertEqual(point1, pointCopy)
+                self.assertFalse(pointCopy != point1)
+                self.assertFalse(point1 != pointCopy)
+            else:
+                self.assertFalse(point1 == point1)
+                self.assertTrue(point1 != point1)
+
+            for lon2, lat2 in self._dataset:
+                if lon1 == lon2 and lat1 == lat2:
+                    continue
+                point2 = SpherePoint(lon2, lat2)
+                self.assertTrue(point2 != point1)
+                self.assertTrue(point1 != point2)
+                self.assertFalse(point2 == point1)
+                self.assertFalse(point1 == point2)
+
+        # Test for transitivity (may be assumed by algorithms).
+        for delta in [10.0**(0.1*x) for x in range(-150, -49, 5)]:
+            self._checkTransitive(delta*radians)
+
+    def _checkTransitive(self, delta):
+        """Test if equality is transitive even for close points.
+
+        This test prevents misuse of approximate floating-point
+        equality -- if `__eq__` is implemented using AFP, then this
+        test will fail for some value of `delta`. Testing multiple
+        values is recommended.
+
+        Parameters
+        ----------
+        delta : `number`
+            The separation, in degrees, at which point equality may
+            become intransitive.
+        """
+        for lon, lat in self._dataset:
+            point1 = SpherePoint(lon - delta, lat)
+            point2 = SpherePoint(lon, lat)
+            point3 = SpherePoint(lon + delta, lat)
+
+            self.assertTrue(point1 != point2 or point2 != point3 or point1 == point3)
+            self.assertTrue(point3 != point1 or point1 != point2 or point3 == point2)
+            self.assertTrue(point2 == point3 or point3 != point1 or point2 == point1)
+
+    def testEqualityAlias(self):
+        """Test if == handles coordinate degeneracies correctly.
+        """
+        # Longitude wrapping
+        self.assertEqual(SpherePoint(360.0*degrees, -42.0*degrees),
+                         SpherePoint(0.0*degrees, -42.0*degrees))
+        self.assertEqual(SpherePoint(-90.0*degrees, -42.0*degrees),
+                         SpherePoint(270.0*degrees, -42.0*degrees))
+
+        # Polar degeneracy
+        self.assertEqual(SpherePoint(42.0*degrees, 90.0*degrees),
+                         SpherePoint(270.0*degrees, 90.0*degrees))
+        self.assertEqual(SpherePoint(-42.0*degrees, -90.0*degrees),
+                         SpherePoint(83.0*degrees, -90.0*degrees))
+
+        self.assertNotEqual(SpherePoint(-42.0*degrees, 90.0*degrees),
+                            SpherePoint(83.0*degrees, -90.0*degrees))
```

Comment:
I suggest you use the same longitude for these two, so that the non-equality is certain to be due to latitude, rather than longitude.